File: app/src/main/java/com/akari/ppx/xp/hook/code/assist/WordHook.java
package com.akari.ppx.xp.hook.code.assist;

import com.akari.ppx.common.utils.XSP;
import com.akari.ppx.xp.hook.code.SuperbHook;

import de.robv.android.xposed.XC_MethodReplacement;

import static com.akari.ppx.common.constant.Prefs.UNLOCK_ILLEGAL_WORD;

public class WordHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		if (!XSP.get(UNLOCK_ILLEGAL_WORD)) return;
		hookMethod("com.sup.android.module.publish.view.NewInputCommentDialog$tryPublish$2", "invoke", String.class, new XC_MethodReplacement() {
			@Override
			protected Object replaceHookedMethod(MethodHookParam param) {
				return param.args[0];
			}
		});
		hookMethod("com.sup.android.m_illegalword.utils.RuleTable", "getReplaceMap", XC_MethodReplacement.returnConstant(null));
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/assist/WordHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.assist

import com.akari.ppx.utils.replaceMethod
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.hook.SwitchHook

class WordHook : SwitchHook("unlock_illegal_words") {
    override fun onHook() {
        "com.sup.android.module.publish.view.NewInputCommentDialog\$tryPublish$2".replaceMethod(
            cl,
            "invoke",
            String::class.java
        ) { param ->
            param.args[0]
        }
        "com.sup.android.m_illegalword.utils.RuleTable".replaceMethod(cl, "getReplaceMap") { null }
    }
}